### PR TITLE
Update sshkey page

### PR DESCRIPTION
**Pull request by samaradel**
### Description
- Remove Active keys table
- Add all keys into one table
- generate a name for keys that haven't a name
- Reduce the size of titles
- prevent opening the dialog of ssh key details by clicking on the active checkbox
- Generate name when the key is imported without name

### Changes

image
### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/2543
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1882

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [x] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by MohamedElmdary — APPROVED** [review]

**Review by AlaaElattar — APPROVED** [review]
Great Job

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]
Good job ya Samar, as we agreed we have to migrate the old key once the user login in background.

image

**Review by MohamedElmdary — APPROVED** [review]

**Review by MohamedElmdary — APPROVED** [review]

**Review by zaelgohary — COMMENTED** [review]
I was able to save the same key more than once. (same name and SSH)

image

File: packages/playground/src/views/sshkey_view.vue
Comment (0oM4R) [inline packages/playground/src/views/sshkey_view.vue:1]: looks like we didn't handle the insufficient balance case
image

**Review by zaelgohary — COMMENTED** [review]
When deleting a key, table IDs won't update.

image

File: packages/playground/src/views/sshkey_view.vue
Comment (0oM4R) [reply packages/playground/src/views/sshkey_view.vue:1]: we need to add error handler also, it keep loading forever

File: packages/playground/src/components/ssh_keys/SshTable.vue [hunk 1]
```diff
@@ -214,20 +206,26 @@ export default defineComponent({
...
     };
 
-    const activateKey = (key: SSHKeyData) => {
-      if (key.isActive) {
-        emit("inactive", key);
-      } else {
-        emit("active", key);
-      }
+    const toggleKeyActivation = (key: SSHKeyData) => {
+      console.log(key.isActive);
```
Comment (0oM4R) [inline packages/playground/src/components/ssh_keys/SshTable.vue:210]: we should remove this line

**Comment by zaelgohary** [comment]
Two buttons have the same label, can we change the first one to 'Export All' instead?

image

**Comment by zaelgohary** [comment]
Header font size doesn't match with other headers.

image

image

**Comment by Mahmoud-Emad** [comment]
> When deleting a key, table IDs won't update.
> 
> image

Shouldn't change, we save it into a database and it's a valid behavior, if we changed the keys I'd each time we delete it'll be a big traffic.

File: packages/playground/src/utils/validators.ts [hunk 2]
```diff
@@ -733,16 +733,6 @@ export function pattern(msg: string, config: RegexPattern) {
...
-/**
- * Checks if a given string is a valid SSH key.
- * @param key - The SSH key string to be validated.
- * @returns True if the key is a valid SSH key, false otherwise.
- */
-export function isValidSSHKey(key: string): boolean {
-  const sshKeyRegex = /^(ssh-rsa|ssh-dss|ecdsa-[a-zA-Z0-9-]+|ssh-ed25519)\s+(\S+)\s+(\S+)/;
-  return sshKeyRegex.test(key);
-}
-
```
Comment (0oM4R) [inline packages/playground/src/utils/validators.ts:745]: why did we move the validation, Shouldn't all validators be here ?

**Comment by zaelgohary** [comment]
Can we change the info to be "Manage SSH keys easily, switch between them, and activate or deactivate keys as needed for accessing deployed machines. Simplify key distribution and effectively manage access to nodes."

https://github.com/threefoldtech/tfgrid-sdk-ts/blob/9ce65149d66df3ae7d1baa8bafe6c6eb8f2b3256/packages/playground/src/views/sshkey_view.vue#L9-L12

**Comment by zaelgohary** [comment]
The lower export button doesn't work as expected.

[export_ssh.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/40770501/720409d2-6e1b-45cf-bc97-70020dd7dcbf)

File: packages/playground/src/utils/validators.ts (hunk 2 above)
Comment (Mahmoud-Emad) [reply packages/playground/src/utils/validators.ts:745]: It moved to be inside the class management, I want to group all ssh-related things in one sope

File: packages/playground/src/views/sshkey_view.vue
Comment (Mahmoud-Emad) [reply packages/playground/src/views/sshkey_view.vue:1]: Resolved

File: packages/playground/src/components/ssh_keys/SshTable.vue (hunk 1 above)
Comment (Mahmoud-Emad) [reply packages/playground/src/components/ssh_keys/SshTable.vue]: Removed

**Review by zaelgohary — APPROVED** [review]
Good job, Emad!

**Review by 0oM4R — CHANGES_REQUESTED** [review]
I still have an unhanded errors on low balance account 
image

File: packages/playground/src/utils/ssh.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+      return;
+    }
+
+    const balance = await loadBalance(grid!);
+    if (balance.free === balance.locked || balance.free < this.updateCost) {
+      throw new InsufficientBalanceError(
+        "Your wallet balance is insufficient to save your SSH key. To avoid losing your SSH key, please recharge your wallet.",
+      );
+    }
+
```
Comment (0oM4R) [inline packages/playground/src/utils/ssh.ts:97]: I think we should wait for the transaction to throw this error and handle it, 
this step should be useful only if we need to disable the btns if the balance is not enough.
what i mean, we should check for the errror if it is an instance of InsufficientBalanceError then create a toast with readable error

**Review by 0oM4R — CHANGES_REQUESTED** [review]
On low balance, an error is handled, but the status has changed!
After reloading the page, the status went back to active. 
also i think the toast should have more friendly message 
[Screencast from 21 أبر, 2024 EET 11:21:09 ص.webm](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/8fb38930-44ac-41d1-8426-7fb8a328c075)

File: packages/playground/src/utils/ssh.ts [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+      return;
+    }
+
+    const balance = await loadBalance(grid!);
+    if (balance.free === balance.locked || balance.free < this.updateCost) {
+      throw new Error(
+        "Your wallet balance is insufficient to save your SSH key. To avoid losing your SSH key, please recharge your wallet.",
+      );
+    }
+
```
Comment (0oM4R) [inline packages/playground/src/utils/ssh.ts:97]: I think there is no need to check for balance and throw errror here, the client already do that for us, 
and this is the idea of using types package and errror classes, client throw an error  then we customize the displayed error massage for the user based on the thrown error class.

File: packages/playground/src/views/sshkey_view.vue [hunk 5]
```diff
@@ -115,258 +100,196 @@
...
+  sshKeysManagement.export(exportKeys);
+  isExporting.value = false;
+};
+
+const updateActivation = async (key: SSHKeyData) => {
+  activating.value = key.activating = true;
+
+  try {
+    await sshKeysManagement.update(allKeys.value);
+  } catch (e: any) {
```
Comment (0oM4R) [inline packages/playground/src/views/sshkey_view.vue:194]: in this catch, we should look up for "insufficientBalanceError" and create a toast with friendly error message, 
and the else block should handle other cases with something like " something went wrong"

File: packages/playground/src/utils/ssh.ts (hunk 4 above)
Comment (Mahmoud-Emad) [reply packages/playground/src/utils/ssh.ts:97]: Yeah, you have a point, errors should be handled outside the management class, However, it is only used in the SSH component and not planned to be used somewhere else, I think we can keep all SSH-related in one place.

File: packages/playground/src/views/sshkey_view.vue (hunk 5 above)
Comment (Mahmoud-Emad) [reply packages/playground/src/views/sshkey_view.vue:194]: The ssh script handle this.

File: packages/playground/src/utils/ssh.ts (hunk 3 above)
Comment (Mahmoud-Emad) [reply packages/playground/src/utils/ssh.ts]: Yeah, indeed.

**Review by 0oM4R — APPROVED** [review]

**Review by 0oM4R — APPROVED** [review]
Thanks ya Omda great job

**Review by Mahmoud-Emad — APPROVED** [review]

**Pull request merged**